src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Services/SyntaxFacts/ISyntaxFacts.cs

```diff
@@ -398,6 +399,12 @@ internal partial interface ISyntaxFacts
         /// </summary>
         string GetNameForArgument(SyntaxNode argument);
 
+        /// <summary>
+        /// Given a <see cref="SyntaxNode"/>, that represents an attribute argument return the string representation of
+        /// that arguments name.
+        /// </summary>
+        string GetNameForAttributeArgument(SyntaxNode argument);
```

Comment:
feel free to add GetAttributeForAttributeArgument as well if you don't want to do the grandparent jump.

Reply:
I just switched to the grandparent jump for now. I guess I was just being too cautious about the possible tree structures.